pkg/pgmodel/querier/query_builder.go

```diff
@@ -334,12 +365,15 @@ func initializeLabeIDMap(labelIDMap map[int64]labels.Label, rows []timescaleRow)
 // getSchemaLabel returns the name and value of schema label if its not empty.
 // Empty value signifies that there is no need to add the schema label.
 // Additionally, we only set the schema if its not the default data schema.
-func getSchemaLabel(metricSchema string) (string, string) {
-	if metricSchema == "" || metricSchema == schema.Data {
-		return "", ""
+func getAdditionalLabels(metricSchema, column string) map[string]string {
```

Comment:
optional: In general returning a small map like this can be expensive (memory allocation+gc pressure). I wonder if we can switch this to a visitor pattern where this runs a passed-in-function for any labels that get added.

Reply:
No really good way to do it because of different types of labels that is used with. Removed map usage and simplified the code for efficiency sake.